src/wallet.cpp

```diff
@@ -1901,11 +1904,26 @@ bool CWallet::CreateCoinStake(const CKeyStore& keystore, unsigned int nBits, int
                 vwtxPrev.push_back(pcoin.first);
                 txNew.vout.push_back(CTxOut(0, scriptPubKeyOut));
 
+                uint64_t nCoinAge;
+                CTxDB txdb("r");
+                if (txNew.GetCoinAge(txdb, nCoinAge))
+                {
+                    uint64_t nTotalSize = pcoin.first->vout[pcoin.second].nValue + GetProofOfStakeReward(nCoinAge, nFees);
+                    if (nTotalSize / 2 > nStakeSplitThreshold * COIN)
```

Comment:
My understanding is that this code will only split the stake if teh total amout is greater that threshold that is set.